### PR TITLE
Integrate product api

### PR DESCRIPTION
**Pull request by manuelkollus**

### CODE REVIEW

File: src/main/java/org/wirvsvirus/locoronando/category/Category.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+@Entity
+public class Category {
+  @Id
+  private UUID uniqueId;
+  private String name;
+  private String description;
+  private int handlerId;
```
Comment (jonasheinisch) [inline src/main/java/org/wirvsvirus/locoronando/category/Category.java]: What is the handlerId?

File: src/main/java/org/wirvsvirus/locoronando/category/Category.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  public UUID uniqueId() {
```
Comment (jonasheinisch) [inline src/main/java/org/wirvsvirus/locoronando/category/Category.java]: @christianor used Lombok to do that. Should we use it globally?

File: src/main/java/org/wirvsvirus/locoronando/category/Category.java (hunk 1 above)
Comment (manuelkollus) [reply src/main/java/org/wirvsvirus/locoronando/category/Category.java]: The handlerId is in this case the dealerId to filter them

File: src/main/java/org/wirvsvirus/locoronando/category/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  @Autowired
+  private CategoryController(
+    CategoryRepository categoryRepository,
+    CategoryService categoryService
+  ) {
+    this.categoryRepository = categoryRepository;
+    this.categoryService = categoryService;
+  }
+
+  @PostMapping(path = "add")
```
Comment (jonasheinisch) [inline src/main/java/org/wirvsvirus/locoronando/category/CategoryController.java]: Why du you add "add" to the path? In classic REST we do not take the action in the path - it's the action (POST) already :)

File: src/main/java/org/wirvsvirus/locoronando/category/Category.java (hunk 2 above)
Comment (manuelkollus) [reply src/main/java/org/wirvsvirus/locoronando/category/Category.java]: I wasn't paying attention, I wouldn't want to use Lombok.

File: src/main/java/org/wirvsvirus/locoronando/category/CategoryController.java (hunk 3 above)
Comment (manuelkollus) [reply src/main/java/org/wirvsvirus/locoronando/category/CategoryController.java]: Oh, okay ;)

File: src/main/java/org/wirvsvirus/locoronando/category/Category.java (hunk 1 above)
Comment (jonasheinisch) [reply src/main/java/org/wirvsvirus/locoronando/category/Category.java]: Don't get it. If it's referencing the dealer, shouldn't we then just reference the Dealer?

**Comment by jonasheinisch** [comment]
@manuelkollus Hab das Zeug aus dem Fuckup-Branch hier reingemerged und mehr oder weniger aufgeräumt. Tut so eigentlich alles. Der ServiceLul ist zum anlegen von Testdaten noch drin. Entweder später rausnehmen oder profilen

**Comment by Tammo0987** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/80f238b8-ac41-4b0c-995b-15c8e367572a?env_id=ibm%3Ayp%3Aeu-de "Backend-Deployment"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/120f8199-f199-4d7d-9c33-7494dd9ff13f "Backend-Delivery") deployed [Locoronando-Backend](https://github.com/Locoronando/Locoronando-Backend/releases/tag/deploy-mycluster-20200322-115222 "deploy-mycluster-20200322-115222") to [mycluster](https://console.bluemix.net/kubernetes/clusters/bpr5p65d0e93e38vprbg/overview "There was a problem retrieving the space name"), including commit 8582a3eeaa02bed67b63d3dba2a9f7bb790372f1

**Pull request merged**